core/src/main/scala/dagr/core/execsystem/TopLikeStatusReporter.scala

```diff
@@ -0,0 +1,214 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2016 Fulcrum Genomics LLC
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package dagr.core.execsystem
+
+import dagr.commons.util.BiMap
+import dagr.commons.util.StringUtil._
+import dagr.commons.util.TimeUtil._
+import dagr.core.tasksystem.{InJvmTask, ProcessTask, Task, UnitTask}
+
+import scala.collection.mutable
+import scala.collection.mutable.ListBuffer
+import scala.tools.jline_embedded.{Terminal => JLineTerminal, TerminalFactory}
+
+/** Simple methods for a terminal */
+object Terminal {
```

Comment:
This feels very weird - because of the way it's implemented you could get a width and a height that are from different points in a resizing operation (e.g. if the user is dragging the window when it's called).  I'd suggest either

```scala
def dimensions: (Int, Int) = Some(terminal).map(t -> (t.getWidth, t.getHeight)).get
```

Or making `Terminal` a `class` that allocates a new `JLineTerminal` each time it's instantiated, and then having `TopLikeStatusReporter` generate a new `Terminal` each time.